client/dashboard/sites/deployments/get-deployment-time-tooltip-text.tsx

```diff
@@ -33,7 +33,7 @@ export const getDeploymentTimeTooltipText = (
 			return sprintf(
 				/* translators: %s is the date and time the deployment finished. e.g. Finished on Sep 19, 2025, 5:11:17 PM */
 				__( 'Finished on %s' ),
-				formatDateTime( completedOn )
+				completedOn ? formatDateTime( completedOn ) : '-'
```

Comment:
I think the bigger problem here is that the type isn't discriminated. If the deployment has finished, then for sure we have the time.

This is a good bandaid but the actual fix would be to create a discriminated type that included the completed timestamp property whenever the status is `success`.

Reply:
Yeah, this needs some refactoring. Because `completedOn` is conditionally non-null, the DeploymentRun type needs to be defined with some type unions I think.

I mainly just wanted to unblock the build for now.

Thank you for approving!